coins/bitcoin/src/rpc.rs

```diff
@@ -0,0 +1,538 @@
+        let info = self.rpc_call::<bitcoin::BlockHash>("getbestblockhash", &[]).await?;
+        Ok(info)
+    }
+
+    pub async fn get_spendable(
+        &self,
+        minconf: Option<usize>,
+        maxconf: Option<usize>,
+        addresses: Option<Vec<&str>>,
+        include_unsafe: Option<bool>,
+    ) -> anyhow::Result<Vec<ListUnspentResultEntry>> {
+        let mut ext_args = [
+            opt_into_json(minconf)?,
+            opt_into_json(maxconf)?,
+            opt_into_json(addresses)?,
+            opt_into_json(include_unsafe)?,
+        ];
+        let defaults = [into_json(1)?, into_json(9999999)?, null(), into_json(true)?];
+        let args = handle_defaults(&mut ext_args, &defaults);
+        let info = self.rpc_call("listunspent", &args).await?;
+        Ok(info)
+    }
+
+    pub async fn get_o_indexes(
+        &self,
+        minconf: Option<usize>,
+        maxconf: Option<usize>,
+        addresses: Option<Vec<&str>>,
+        include_unsafe: Option<bool>,
+    ) -> anyhow::Result<Vec<ListUnspentResultEntry>> {
+        let info: Vec<ListUnspentResultEntry> = self
+            .get_spendable(minconf, maxconf, addresses, include_unsafe)
+            .await?;
+        Ok(info)
+    }
+
+    pub async fn lock_unspent(&self, outputs: &[OutPoint]) -> Result<bool> {
+        let outputs: Vec<_> = outputs
+            .iter()
+            .map(|o| serde_json::to_value(JsonOutPoint::from(*o)).unwrap())
+            .collect();
+        let info: bool = self
+            .rpc_call::<bool>("lockunspent", &[false.into(), outputs.into()])
+            .await?;
+        Ok(info)
+    }
+
+    pub async fn unlock_unspent(&self, outputs: &[OutPoint]) -> Result<bool> {
+        let outputs: Vec<_> = outputs
+            .iter()
+            .map(|o| serde_json::to_value(JsonOutPoint::from(*o)).unwrap())
+            .collect();
+        let mut ext_args = [true.into(), outputs.into()];
+        let args = handle_defaults(&mut ext_args, &[true.into(), empty_arr()]);
+        let info: bool = self.rpc_call::<bool>("lockunspent", &args).await?;
+        Ok(info)
+    }
+
+    pub async fn get_fee_per_byte(&self, conf_target: Option<usize>) -> anyhow::Result<u64> {
+        let mut ext_args = [opt_into_json(conf_target)?];
+        let defaults = [into_json(100)?];
+        let args = handle_defaults(&mut ext_args, &defaults);
+        let fee: EstimateSmartFeeResult = self
+            .rpc_call::<EstimateSmartFeeResult>("estimatesmartfee", &args)
+            .await?;
+        Ok(fee.fee_rate.unwrap().to_sat())
+    }
+
+    pub async fn get_transaction(&self, tx_hash: &str) -> anyhow::Result<GetTransactionResult> {
+        let mut ext_args = [into_json(tx_hash)?];
+        let args = handle_defaults(&mut ext_args, &[null()]);
+        let tx = self
+            .rpc_call::<GetTransactionResult>("gettransaction", &args)
+            .await?;
+        Ok(tx)
+    }
+
+    pub async fn get_transactions(
```

Comment:
Remove get_transaction and get_transactions for functions (such as the following get_raw_transactions) supporting `bitcoin`'s Transaction, a usable and portable type.